lib/buffer.js

```diff
@@ -537,15 +537,44 @@ Buffer.prototype.inspect = function inspect() {
   return '<' + this.constructor.name + ' ' + str + '>';
 };
 
+Buffer.prototype.compare = function compare(target,
+                                            start,
+                                            end,
+                                            thisStart,
+                                            thisEnd) {
 
-Buffer.prototype.compare = function compare(b) {
-  if (!(b instanceof Buffer))
+  if (!(target instanceof Buffer))
     throw new TypeError('Argument must be a Buffer');
 
-  if (this === b)
+  if (typeof start === 'undefined')
+    start = 0;
+  if (typeof end === 'undefined')
+    end = target ? target.length : 0;
+  if (typeof thisStart === 'undefined')
+    thisStart = 0;
+  if (typeof thisEnd === 'undefined')
+    thisEnd = this.length;
```

Comment:
Nice. For reference, this should operate the same way as typed arrays:
```js
new Uint8Array(new ArrayBuffer(8), undefined, 5).byteOffset === 0
new Uint8Array(new ArrayBuffer(8), false, 5).byteOffset === 0
new Uint8Array(new ArrayBuffer(8), true, 5).byteOffset === 1
new Uint8Array(new ArrayBuffer(8), 0, undefined).byteLength === 0
new Uint8Array(new ArrayBuffer(8), 0, null).byteLength === 0
new Uint8Array(new ArrayBuffer(8), 0, true).byteLength === 1
```
Do you believe these existing tests covers this?

I may recommend dropping the `typeof`. Not important enough for me to block, but not sure there's a need for the extra keyword.